cyclops-ctrl/internal/cluster/k8sclient/modules.go

```diff
@@ -471,6 +472,54 @@ func (k *KubernetesClient) getStatefulsetPods(deployment appsv1.StatefulSet) ([]
 	return out, nil
 }
 
+func (k *KubernetesClient) getPodsForCronJob(cronJob batchv1.CronJob) ([]dto.Pod, error) {
+	jobTemplateLabels := cronJob.Spec.JobTemplate.Spec.Template.Labels
+	labelSelector := labels.SelectorFromSet(jobTemplateLabels).String()
+	pods, err := k.clientset.CoreV1().Pods(cronJob.Namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Since cronjobs don't create pods directly but create jobs that then create pods, you will need to fetch jobs for a cronjob first and then fetch all the pods for each job.

```go
	jobTemplateLabels := cronJob.Spec.JobTemplate.Spec.Template.Labels
	jobLabelSelector := labels.SelectorFromSet(jobTemplateLabels).String()

	jobs, err := k.clientset.BatchV1().Jobs(cronJob.Namespace).List(context.Background(), metav1.ListOptions{
		LabelSelector: jobLabelSelector,
	})
	if err != nil {
		return nil, err
	}

	out := make([]dto.Pod, 0, 0)
	for _, job := range jobs.Items {
		podTemplateLabels := job.Spec.Template.Labels
		podLabelSelector := labels.SelectorFromSet(podTemplateLabels).String()
                .
                .
                .
         }
         return out, nil
```